cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/AbstractStreamWriterSizeManager.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data.stream.service;
+
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.data.stream.service.heartbeat.HeartbeatPublisher;
+import co.cask.cdap.data.stream.service.heartbeat.StreamWriterHeartbeat;
+import com.google.common.collect.Maps;
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.common.util.concurrent.ListeningScheduledExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.inject.name.Named;
+import org.apache.twill.common.Threads;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Common implementation of a {@link StreamWriterSizeManager} that keeps the size of data ingested by one stream writer
+ * for all streams in memory.
+ */
+public abstract class AbstractStreamWriterSizeManager
+  extends AbstractIdleService
```

Comment:
Actually, I was looking into it, and since the scheduling concerns each stream, and not the entire service, we can't simply use one scheduled method. Unless we schedule one method to send a heartbeat for all existing streams at once. What do you think is better?